packages/grid/src/key-handlers/SelectionKeyHandler.ts

```diff
@@ -194,9 +194,12 @@ class SelectionKeyHandler extends KeyHandler {
 
         grid.moveCursorToPosition(top, cursorColumn, isShiftKey, false);
 
-        grid.setViewState({ top });
+        grid.setViewState({ top }, false, event);
       } else {
+        const { lastLeft, lastTop } = grid.metrics;
+
         grid.moveCursor(deltaColumn, deltaRow, isShiftKey);
+        grid.setViewState({ left: lastLeft, top: lastTop }, false, event); // need to call setViewState again since moveCursor calls a series of functions that sets view state
```

Comment:
`lastLeft` and `lastTop` is the furthest point that can be scrolled - with this change trying to navigate with the keyboard always jumps to the bottom.
We're probably just going to have to pass the event through `grid.moveCursor` and the other methods it has to go through, since we don't actually know where the cursor/viewport ends up until the state is set (that's done in `grid.moveViewToCell`).